src/v/datalake/translation/partition_translator.cc

```diff
@@ -24,12 +25,19 @@
 #include "datalake/translation/state_machine.h"
 #include "datalake/translation_task.h"
 #include "kafka/utils/txn_reader.h"
+#include "model/fundamental.h"
+#include "model/timeout_clock.h"
 #include "resource_mgmt/io_priority.h"
 #include "ssx/future-util.h"
 #include "utils/lazy_abort_source.h"
 
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/future.hh>
 #include <seastar/coroutine/as_future.hh>
 
+#include <functional>
```

Comment:
please remove unused includes

Reply:
Which include do you think is unused?

`<functional>` is used [here](https://github.com/redpanda-data/redpanda/pull/24780/commits/7d90b87b99019ad32b74cfb1af5f11a339ba9682#diff-617058a8220c70fdc909b169e594470f7b81ed70497f71c95f421c9da374aee8R160).